src/equation_systems/equation_system.C

```diff
@@ -122,9 +163,9 @@ EquationSystem::FormLinearSystem(mfem::OperatorHandle & op,
     auto blf = _blfs.Get(test_var_name);
     auto lf = _lfs.Get(test_var_name);
     mfem::Vector aux_x, aux_rhs;
-    _h_blocks(i, i) = new mfem::HypreParMatrix;
-    blf->FormLinearSystem(
-        _ess_tdof_lists.at(i), *(_xs.at(i)), *lf, *_h_blocks(i, i), aux_x, aux_rhs);
+    mfem::HypreParMatrix aux_a;
+    blf->FormLinearSystem(_ess_tdof_lists.at(i), *(_xs.at(i)), *lf, aux_a, aux_x, aux_rhs);
+    _h_blocks(i, i) = new const mfem::HypreParMatrix(aux_a);
```

Comment:
Worth adding a comment here about intended ownership of the new mfem::HypreParMatrix

Reply:
Done!